intranet/apps/preferences/fields.py

```diff
@@ -10,22 +10,23 @@ class PhoneField(forms.Field):
         'invalid': 'Please enter a valid phone number.'
     }
 
-    def __init__(self, max_length=None, *args, **kwargs):
-        self.max_length = max_length
+    def __init__(self, *args, **kwargs):
         super(PhoneField, self).__init__(*args, **kwargs)
-        self.validators.append(validators.RegexValidator(r'^[0-9]+$', 'Please enter a valid phone number.'))
-        if max_length is not None:
-            self.validators.append(validators.MaxLengthValidator(int(max_length)))
+        self.validators.append(validators.RegexValidator(r'^\+?1?\d{9,15}$', 'Please enter a valid phone number.'))
 
     def to_python(self, value):
         "Returns a Unicode object."
         if value in self.empty_values:
-            return ''
+            return ""
         value = force_text(value).strip()
         return value
 
+    def prepare_value(self, value):
+        if value == "None":
+            return ""
+        return value
+
     def widget_attrs(self, widget):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/tjcsl/ion/file/2365667582/issues/source?bid=3176160&fileBranchId=3221080#l29)